compiler/src/dotty/tools/dotc/transform/TreeChecker.scala

```diff
@@ -122,6 +122,10 @@ class TreeChecker extends Phase with SymTransformer {
     val squahsedPhase = ctx.squashed(prevPhase)
     ctx.echo(s"checking ${ctx.compilationUnit} after phase ${squahsedPhase}")
 
+    if (ctx.isAfterTyper) {
```

Comment:
Yes, this is the right place.
TreeChecker is always after typer, no need for the if.